src/print.jl

```diff
@@ -324,6 +326,84 @@ function objective_function_string(print_mode, model::Model)
    end
 end
 
+"""
+    show_solution_summary([io::IO], model::Model; verbose::Bool = false)
+
+Write to `io` (or to the default output stream `stdout` if `io` is not given)
+a summary of the solution results. 
+If `verbose` is true, additional information are written.
+"""
+function show_solution_summary(io::IO, model::Model; verbose::Bool = false)
+    println(io, "* Solver : ", solver_name(model))
+    println(io, "")
+
+    _show_status_summary(io, model, verbose)
+    _show_candidate_solution_summary(io, model, verbose)
+    _show_work_counters_summary(io, model, verbose)
+    return
+end
+
+show_solution_summary(model::Model; verbose::Bool = false) = show_solution_summary(stdout, model; verbose = verbose)
+
+function _show_status_summary(io::IO, model::Model, verbose::Bool)
+    println(io, "* Status")
+    println(io, "  Termination status : ", termination_status(model))
+    println(io, "  Primal status      : ", primal_status(model))
+    println(io, "  Dual status        : ", dual_status(model))
+    verbose && println(io, "  Result count       : ", result_count(model))
+    verbose && println(io, "  Has values         : ", has_values(model))
+    verbose && println(io, "  Has duals          : ", has_duals(model))
+    println(io, "  Message from the solver:")
+    println(io, "  \"", raw_status(model), "\"")
+    println(io, "")
+end
+
+function _show_candidate_solution_summary(io::IO, model::Model, verbose::Bool)
+    println(io, "* Candidate solution")
+    println(io, "  Objective value      : ", objective_value(model))
+    _try_print(io, "  Objective bound      : ", objective_bound, model)
+    _try_print(io, "  Dual objective value : ", dual_objective_value, model)
+
+    if verbose && has_values(model)
+        println(io, "  Primal solution : ")
+        for x in all_variables(model)
+            println(io, "    ", name(x), " : ", value(x))
+        end
+    end
+
+    if verbose && has_duals(model)
+        println(io, "  Dual solution : ")
+        constraint_id = 1
+        for (F, S) in list_of_constraint_types(model)
+            for constraint in all_constraints(model, F, S)
+                constraint_name = name(constraint)
+                if isempty(constraint_name) 
```

Comment:
I wonder if we should skip constraints (and variables) that have empty names. If they aren't named, they probably aren't important to the user. And printing things with temporary names isn't very helpful because the user won't know what they refer to. What do you think?

Reply:
I agree. Maybe the temporary name is not useful in usual cases. I fixed it.